### ISSUE TITLE
False positive: match_same_arms const

### ISSUE THREAD

**Issue by alexheretic** [rust-lang-nursery/rust-clippy#2427]
Similar to #2420, just because a const is the same value _this compilation_ doesn't mean it's the same thing.
```rust
const PRICE_OF_SWEETS: u32 = 5;
const PRICE_OF_KINDNESS: u32 = 0;
const PRICE_OF_DRINKS: u32 = 5;

fn price(thing: &str) -> u32 {
    match thing {
        "rolo" => PRICE_OF_SWEETS,
        "advice" => PRICE_OF_KINDNESS,
        "juice" => PRICE_OF_DRINKS,
        _ => panic!()
    }
}
```
```
warning: this `match` has identical arm bodies
  --> src/main.rs:16:20
   |
16 |         "juice" => PRICE_OF_DRINKS,
   |                    ^^^^^^^^^^^^^^^
   |
   = note: #[warn(match_same_arms)] on by default
note: same as this
  --> src/main.rs:14:19
   |
14 |         "rolo" => PRICE_OF_SWEETS,
   |                   ^^^^^^^^^^^^^^^
note: consider refactoring into `"rolo" | "juice"`
  --> src/main.rs:14:19
   |
14 |         "rolo" => PRICE_OF_SWEETS,
   |                   ^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.184/index.html#match_same_arms
```

**Comment by oli-obk** [comment]
Our const evaluator has a bool flag that makes it treat constants as black boxes. We just need to turn it on for this check.

**Comment by 8573** [comment]
Are this and #2185 duplicates?
